netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -105,6 +128,59 @@ public void onGoAwayReceived(int lastStreamId, long errorCode, ByteBuf debugData
     });
   }
 
+  /**
+   * Sends a ping to the server.
+   */
+  void ping(final PingCallback callback, Executor executor) {
```

Comment:
Yeah ... that's the case for all callbacks with `directExecutor` :)  But I don't suspect that handling a ping will be that taxing.